From ead82362a2529d559fc9b697dedef2f5751165a7 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Sun, 01 Aug 2021 16:13:12 +0300
Subject: [PATCH] [SW-49907] gaudi: ras: fix btl index

Fix BTL index reading for RAS, as the BTL ID is
in LSB byte, not MSB. Later, in u-boot it is shifted
8 bytes left, which explains why in Linux we retrieve
BTL index from MSB byte.

Change-Id: I6f12e1874c02f784d6230fc19e912ff4b27b5102
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/arch/gaudi/asic.c b/arch/gaudi/asic.c
index d3e381d..2f0955d 100644
--- a/arch/gaudi/asic.c
+++ b/arch/gaudi/asic.c
@@ -466,7 +466,7 @@
 
 u32 hl_get_btl_index(void)
 {
-	u32 btl_id = (regmap_readl(mmBTL_ID) >> 8) & 0xff;
+	u32 btl_id = regmap_readl(mmBTL_ID) & 0xff;
 
 	return (btl_id & 0x3) - 1;
 }
